File: src/main/java/com/dashboard/controllers/MessageController.java
//package com.dashboard.controllers;
//
//import com.dashboard.interfaces.controllerInterfaces.MessageControllerInterface;
//import com.dashboard.models.JPA.Message;
//import lombok.extern.log4j.Log4j2;
//import org.springframework.http.HttpStatus;
//import org.springframework.http.ResponseEntity;
//import org.springframework.web.bind.annotation.RequestMapping;
//import org.springframework.web.bind.annotation.RestController;
//
//@Log4j2
//@RestController
//@RequestMapping("/api/v1/messages")
//public class MessageController implements MessageControllerInterface {
//
//    @Override
//    public ResponseEntity<?> createMessage(Message message) {
//        return new ResponseEntity<>("NOT IMPLEMENTED", HttpStatus.NOT_IMPLEMENTED);
//    }
//
//    @Override
//    public ResponseEntity<?> getAllMessages() {
//        return new ResponseEntity<>("NOT IMPLEMENTED", HttpStatus.NOT_IMPLEMENTED);
//    }
//
//    @Override
//    public ResponseEntity<?> getMessageById(Integer messageID) {
//        return new ResponseEntity<>("NOT IMPLEMENTED", HttpStatus.NOT_IMPLEMENTED);
//    }
//
//    @Override
//    public ResponseEntity<?> deleteMessageById(Integer messageID) {
//        return new ResponseEntity<>("NOT IMPLEMENTED", HttpStatus.NOT_IMPLEMENTED);
//    }
//
//    @Override
//    public ResponseEntity<?> updateMessage(Integer messageID, Message Message) {
//        return new ResponseEntity<>("NOT IMPLEMENTED", HttpStatus.NOT_IMPLEMENTED);
//    }
//}

File: src/main/kotlin/api/MessageController.kt
package api

import interfaces.controllerInterfaces.MessageControllerInterface
import lombok.extern.log4j.Log4j2
import models.jpa.Message
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.RestController
import org.springframework.web.bind.annotation.RequestMapping


@RestController
@RequestMapping("/api/v1/messages")
class MessageController : MessageControllerInterface {

    override fun createMessage(message: Message): ResponseEntity<*> {
        return ResponseEntity("NOT IMPLEMENTED", HttpStatus.NOT_IMPLEMENTED)
    }

    override fun allMessages(): ResponseEntity<*> = ResponseEntity("NOT IMPLEMENTED", HttpStatus.NOT_IMPLEMENTED)

    override fun getMessageById(messageID: Int): ResponseEntity<*> {
        return ResponseEntity("NOT IMPLEMENTED", HttpStatus.NOT_IMPLEMENTED)
    }

    override fun deleteMessageById(messageID: Int): ResponseEntity<*> {
        return ResponseEntity("NOT IMPLEMENTED", HttpStatus.NOT_IMPLEMENTED)
    }

    override fun updateMessage(messageID: Int, Message: Message): ResponseEntity<*> {
        return ResponseEntity("NOT IMPLEMENTED", HttpStatus.NOT_IMPLEMENTED)
    }
}